invokeai/app/api/routers/images.py

```diff
@@ -377,14 +378,70 @@ class ImagesDownloaded(BaseModel):
     response: Optional[str] = Field(
         description="If defined, the message to display to the user when images begin downloading"
     )
+    bulk_download_item_name: str = Field(description="The bulk download item name of the bulk download item")
 
 
-@images_router.post("/download", operation_id="download_images_from_list", response_model=ImagesDownloaded)
+@images_router.post(
+    "/download", operation_id="download_images_from_list", response_model=ImagesDownloaded, status_code=202
+)
 async def download_images_from_list(
-    image_names: list[str] = Body(description="The list of names of images to download", embed=True),
+    background_tasks: BackgroundTasks,
+    image_names: Optional[list[str]] = Body(
+        default=None, description="The list of names of images to download", embed=True
+    ),
     board_id: Optional[str] = Body(
         default=None, description="The board from which image should be downloaded from", embed=True
     ),
 ) -> ImagesDownloaded:
-    # return ImagesDownloaded(response="Your images are downloading")
-    raise HTTPException(status_code=501, detail="Endpoint is not yet implemented")
+    if (image_names is None or len(image_names) == 0) and board_id is None:
+        raise HTTPException(status_code=400, detail="No images or board id specified.")
+    bulk_download_item_id: str = uuid_string() if board_id is None else board_id
+    board_name: str = (
+        "" if board_id is None else ApiDependencies.invoker.services.bulk_download.get_clean_board_name(board_id)
+    )
+
+    # Type narrowing handled above ^, we know that image_names is not None, trying to keep null checks at the boundaries
+    background_tasks.add_task(
+        ApiDependencies.invoker.services.bulk_download.handler,
+        cast(list[str], image_names),
+        board_id,
+        bulk_download_item_id,
+    )
+    return ImagesDownloaded(
+        response="Your images are preparing to be downloaded",
+        bulk_download_item_name=bulk_download_item_id if board_id is None else board_name + ".zip",
```

Comment:
This would not work, the if would not include .zip when board_id is not provided